ledger/participant-integration-api/src/main/scala/platform/store/dao/EventProjectionProperties.scala

```diff
@@ -0,0 +1,97 @@
+  *  interfaces and templates implementing them.
+  *
+  * @param verbose enriching in verbose mode
+  * @param witnessTemplateIdFilter populate contract_argument, and contract_key. If templateId set is empty: populate.
+  * @param witnessInterfaceViewFilter populate interface_views. The Map of templates to interfaces,
+  *                              and the set of implementor templates cannot be empty.
+  */
+final case class EventProjectionProperties private[dao] (
+    verbose: Boolean,
+    // Map(eventWitnessParty, Set(templateId))
+    witnessTemplateIdFilter: Map[String, Set[Identifier]] = Map.empty,
+    // Map(eventWitnessParty, Map(templateId -> Set(interfaceId)))
+    witnessInterfaceViewFilter: Map[String, Map[Identifier, Set[Identifier]]] = Map.empty,
+) {
+  def render(witnesses: Set[String], templateId: Identifier): RenderResult = {
+    val renderContractArguments: Boolean = witnesses.view
+      .flatMap(witnessTemplateIdFilter.get)
+      .exists(templates => templates.isEmpty || templates(templateId))
+
+    val interfacesToRender: Set[Identifier] = witnesses.view
+      .flatMap(witnessInterfaceViewFilter.get(_).iterator)
+      .flatMap(_.getOrElse(templateId, Set.empty[Identifier]))
+      .toSet
+
+    RenderResult(renderContractArguments, interfacesToRender)
+  }
+
+}
+
+object EventProjectionProperties {
+
+  case class RenderResult(
+      contractArguments: Boolean,
+      interfaces: Set[Identifier],
+  )
+
+  /** @param transactionFilter Transaction filter as defined by the consumer of the API.
+    * @param verbose                 enriching in verbose mode
+    * @param interfaceImplementedBy  The relation between an interface id and template id.
+    *                                If template has no relation to the interface,
+    *                                an empty Set must be returned.
+    */
+  def apply(
+      transactionFilter: domain.TransactionFilter,
+      verbose: Boolean,
+      interfaceImplementedBy: Identifier => Set[Identifier],
+  ): EventProjectionProperties =
+    EventProjectionProperties(
+      verbose = verbose,
+      witnessTemplateIdFilter = witnessTemplateIdFilter(transactionFilter),
+      witnessInterfaceViewFilter =
+        witnessInterfaceViewFilter(transactionFilter, interfaceImplementedBy),
+    )
+
+  private def witnessTemplateIdFilter(
+      domainTransactionFilter: domain.TransactionFilter
+  ): Map[String, Set[Identifier]] =
+    domainTransactionFilter.filtersByParty.iterator
+      .map { case (party, filters) => (party.toString, filters) }
+      .collect {
+        case (party, Filters(None)) => party -> Set.empty[Identifier]
+        case (party, Filters(Some(empty)))
+            if empty.templateIds.isEmpty && empty.interfaceFilters.isEmpty =>
+          party -> Set.empty[Identifier]
+        case (party, Filters(Some(nonEmptyFilter))) if nonEmptyFilter.templateIds.nonEmpty =>
```

Comment:
Nice simplification. There is one thing that's not clear to me: if `Filters(Some(filter))` `filter.interfaceFilters.nonEmpty` and `filter.templateIds.isEmpty` then it is assumed that the empty `templateIds`  was not meant as a wildcard, which is not very uniform. Am I missing something? @nmarton-da 

@skisel-da Don't block the PR on this question. It's just for my understanding. Also asked on the interface changed PR as well: https://github.com/digital-asset/daml/pull/14033/files#r942612307

Reply:
Got an answer there